jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/api/ConstraintBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Two Sigma Open Source, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twosigma.cook.jobclient.constraint.api;
+
+import org.json.JSONArray;
+
+public interface ConstraintBuilder {
```

Comment:
I imagined the constraint builder to have factory methods like:
`Constraint<String> buildEqualsConstraint(String, String)` and `Constraint<String[]> buildInConstraint(String, String[])`

Reply:
How about the UNIQUE operator?